pkg/daemon/bootc.go

```diff
@@ -0,0 +1,245 @@
+package daemon
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"time"
+
+	"k8s.io/klog/v2"
+)
+
+// bootc-client-go.go
+// BootcStatus summarizes the current worldview of the bootc status --json.
+type BootcStatus struct {
+	APIVersion string   `json:"apiVersion"`
+	Kind       string   `json:"kind"`
+	Metadata   Metadata `json:"metadata"`
+	Spec       Spec     `json:"spec"`
+	Status     Status   `json:"status"`
+}
+
+type Metadata struct {
+	Name string `json:"name"`
+}
+
+// Spec collects the host specification
+// This mirrors https://github.com/containers/bootc/blob/94521ecb19b8d0f2bfc36220322c60cafa036296/lib/src/spec.rs#L46
```

Comment:
Agree!! I think currently we are doing it manually for both bootc bindings and rpm-ostree bindings. It will def be better to have something auto-gen. Colin might have some idea according to comment down: https://issues.redhat.com/browse/MCO-1026 ?

Reply:
https://github.com/containers/bootc/pull/703 - does that help?

Reply:
Yes, that would help! We'll need a way to consume that JSON schema, but it's a step in the right direction.

Reply:
Some trials on autogen-ing bootc go-bindings by consuming the JSON schema: https://github.com/inesqyx/bootc-client-go